crypto/x509/x509_lu.c

```diff
@@ -322,19 +322,17 @@ int X509_STORE_add_cert(X509_STORE *ctx, X509 *x)
     CRYPTO_THREAD_write_lock(ctx->lock);
 
     if (X509_OBJECT_retrieve_match(ctx->objs, obj)) {
-        X509err(X509_F_X509_STORE_ADD_CERT,
-                X509_R_CERT_ALREADY_IN_HASH_TABLE);
-        ret = 0;
+        ret = 1;
     } else {
         added = sk_X509_OBJECT_push(ctx->objs, obj);
         ret = added != 0;
     }
 
     CRYPTO_THREAD_unlock(ctx->lock);
 
-    if (!ret)                   /* obj not pushed */
+    if (!added)                 /* obj not pushed */
```

Comment:
I'm sorry, a code style nit pick: since `added` isn't purely boolean, that should be `if (added == 0)`